kikuchipy/signals/tests/test_ebsd_masterpattern.py

```diff
@@ -166,3 +180,136 @@ def test_set_phase_parameters(self):
         md_dict = s.metadata.get_item("Sample.Phases.1").as_dictionary()
         p.pop("number")
         assert_dictionary(p, md_dict)
+
+
+class TestEBSDCatalogue:
+
+    # Create  detector model
+    detector = EBSDDetector(
+        shape=(480, 640),
+        px_size=50,
+        pc=(20, 20, 15000),
+        convention="emsoft4",
+        tilt=10,
+        sample_tilt=70,
+    )
+
+    def test_get_direction_cosines(self):
+        out = _get_direction_cosines(self.detector)
+        assert isinstance(out, Vector3d)
+
+    def test_get_lambert_interpolation_parameters(self):
+        dc = _get_direction_cosines(self.detector)
+        scl = 500
+        npx = 1001
+        npy = 1001
+        (
+            nii,
+            nij,
+            niip,
+            nijp,
+            di,
+            dj,
+            dim,
+            djm,
+        ) = _get_lambert_interpolation_parameters(dc, scl, npx, npy)
+
+        assert (nii <= niip).all()
+        assert (nij <= nijp).all()
+
+        assert (nii < npx).all()
+        assert (nij < npy).all()
+        assert (niip < npx).all()
+        assert (nijp < npx).all()
+
+        assert (nii >= 0).all()
+        assert (nij >= 0).all()
+        assert (niip >= 0).all()
+        assert (nijp >= 0).all()
+
+    def test_get_patterns(self):
+
+        # Ni Test
+        EMSOFT_EBSD_FILE = os.path.join(
```

Comment:
As discussed off GitHub, this file should be part of the kikuchipy.data module. I can add that after this PR is in. You can resolve this comment when you want to.